framework/web/Request.php

```diff
@@ -1477,6 +1493,181 @@ protected function loadCookies()
+                'class' => $this->uploadedFileClass,
+                'clientFilename' => $names,
+                'tempFilename' => $tempNames,
+                'clientMediaType' => $types,
+                'size' => $sizes,
+                'error' => $errors,
+            ]);
+        }
+    }
+
+    /**
+     * Returns an uploaded file according to the given name.
+     * Name can be either a string HTML form input name, e.g. 'Item[file]' or array path, e.g. `['Item', 'file']`.
+     * Note: this method returns `null` in case given name matches multiple files.
+     * @param string|array $name HTML form input name or array path.
+     * @return UploadedFileInterface|null uploaded file instance, `null` - if not found.
+     * @since 2.1.0
+     */
+    public function getUploadedFileByName($name)
+    {
+        $uploadedFile = $this->findUploadedFiles($name);
+        if ($uploadedFile instanceof UploadedFileInterface) {
+            return $uploadedFile;
+        }
+        return null;
+    }
+
+    /**
+     * Returns the list of uploaded file instances according to the given name.
+     * Name can be either a string HTML form input name, e.g. 'Item[file]' or array path, e.g. `['Item', 'file']`.
+     * Note: this method does NOT preserve uploaded files structure - it returns instances in single-level array (list),
+     * even if they are set by nested keys.
+     * @param string|array $name HTML form input name or array path.
+     * @return UploadedFileInterface[] list of uploaded file instances.
+     * @since 2.1.0
+     */
+    public function getUploadedFilesByName($name)
+    {
+        $uploadedFiles = $this->findUploadedFiles($name);
+        if ($uploadedFiles === null) {
+            return [];
+        }
+        if ($uploadedFiles instanceof UploadedFileInterface) {
+            return [$uploadedFiles];
+        }
+        return $this->reduceUploadedFiles($uploadedFiles);
+    }
+
+    /**
+     * Returns an uploaded file for the given model attribute.
+     * @param \yii\base\Model $model the data model.
+     * @param string $attribute the attribute name. The attribute name may contain array indexes.
+     * For example, '[1]file' for tabular file uploading; and 'file[1]' for an element in a file array.
+     * @return UploadedFileInterface|null uploaded file instance, `null` - if not found.
+     * @since 2.1.0
+     */
+    public function getUploadedFileByModel($model, $attribute)
+    {
+        return $this->getUploadedFileByName(Html::getInputName($model, $attribute));
+    }
+
+    /**
+     * Returns all uploaded files for the given model attribute.
+     * Note: this method does NOT preserve uploaded files structure - it returns instances in single-level array (list),
+     * even if they are set by nested keys.
+     * @param \yii\base\Model $model the data model
+     * @param string $attribute the attribute name. The attribute name may contain array indexes
+     * for tabular file uploading, e.g. '[1]file'.
```

Comment:
Isn't this sample supposed to be `file[1]`?

Reply:
No, not in this case. It's not 1st of files, it's file for 1st of records we're filling in.

Reply:
Ah, I see.